app/src/main/java/com/example/superimagesviewer/ui/InstagramPostFragment.kt

```diff
@@ -25,16 +29,23 @@ class InstagramPostFragment : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         editTextPhotoUrl = view.findViewById(R.id.edit_text_photo_url)
         uploadButton = view.findViewById(R.id.upload_button)
-    }
+        mosaicPageButton = view.findViewById(R.id.post_to_mosaic_page_button)
 
-    override fun onActivityCreated(savedInstanceState: Bundle?) {
-        super.onActivityCreated(savedInstanceState)
-        instagramPostViewModel = ViewModelProvider(this).get(InstagramPostViewModel::class.java)
+        mosaicPageButton.setOnClickListener{
+           view.findNavController().navigate(
+               InstagramPostFragmentDirections.actionInstagramPostFragmentToMosaicFragment()
+           )
+        }
 
         uploadButton.setOnClickListener {
-            instagramPostViewModel.mosaicRepository.uploadPhotoToInstagramByUrl(editTextPhotoUrl.text.toString())
+            instagramPostViewModel.photoUrl = editTextPhotoUrl.text.toString()
```

Comment:
Btw, it is possible that user enter invalid or empty url, please add validation for this scenario

Reply:
done